examples/siro_sandbox/sandbox_app.py

```diff
@@ -1059,15 +1112,7 @@ def sim_update(self, dt):
             self._update_task()
             self._update_grasping_and_set_act_hints()
 
-        # Navmesh visualization only works in the debug third-person view
-        # (--debug-third-person-width), not the main sandbox viewport. Navmesh
-        # visualization is only implemented for simulator-rendering, not replay-
-        # rendering.
-        if self.gui_input.get_key_down(GuiInput.KeyNS.N):
-            self.env._sim.navmesh_visualization = (  # type: ignore
-                not self.env._sim.navmesh_visualization  # type: ignore
-            )
-
+        self._add_spatial_data_to_gfx_replay_keyframe()
```

Comment:
In the scenario where the tutorial is running or the episode is done/inactive, this will keep adding these user transforms over and over again to the last keyframe in `gfx_replay_manager`. This feels weird. Maybe we should only call this in `_check_compute_action_and_step_env`, after actually stepping the env.

Reply:
@eundersander  Do we want gfx-replay keyframes to be saved only for the  `self._sim_update_controlling_agent(dt)` or for `self._sim_update_tutorial(dt)` too?

Reply:
We can move `self._add_spatial_data_to_gfx_replay_keyframe()` to `_check_compute_action_and_step_env ` if we want saving keyframes from `self._sim_update_controlling_agent(dt)` only.